src/MassTransit.Abstractions/Configuration/DependencyInjection/IRegistrationContext.cs

```diff
@@ -8,6 +8,7 @@ namespace MassTransit
     /// receive endpoints.
     /// </summary>
     public interface IRegistrationContext :
+        ISetScopedConsumeContext,
```

Comment:
I do not like this, we should not expose `PushContext` to end-users during configuration.